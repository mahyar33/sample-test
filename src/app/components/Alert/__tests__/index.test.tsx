import React from 'react';
import { render } from '@testing-library/react';

import { Alert } from '../index';

const renderWithProps = (props: Parameters<typeof Alert>[number]) =>
  render(<Alert {...props} />);

describe('<Alert />', () => {
  it('should render an <div> tag', () => {
    const alert = renderWithProps({ textAlert: 'hi' });
    expect(alert.container.querySelector('div')).toBeInTheDocument();
  });
  it('should match snapshot', () => {
    const alert = renderWithProps({ textAlert: 'hi' });
    expect(alert.container.firstChild).toMatchSnapshot();
  });
});
