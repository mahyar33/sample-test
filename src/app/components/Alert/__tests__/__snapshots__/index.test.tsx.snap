// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> should match snapshot 1`] = `
.c0 {
  color: white;
  background-color: #e43c3c;
  padding: 0.75rem 1.25rem;
  margin-bottom: 1rem;
  border-radius: 0.25rem;
  font-weight: bold;
  margin: 2rem 0;
}

<div
  class="c0"
  role="alert"
>
  hi
</div>
`;
